### ISSUE TITLE
How do you implement the equation(15) in your paper?

### ISSUE THREAD

**Issue by Dorispaopao** [XiaLiPKU/EMANet#5]
And how to consider the gradient backpropgation in your implement?

**Comment by XiaLiPKU** [comment]
For the first question:
I implemented it in the 'train.py' https://github.com/XiaLiPKU/EMANet/blob/9a492d8aaad297e15eac044b3bb9583e63ffa3a3/train.py#L134. Implement it in the EMAU module may be more good-looking. But as the \mu has to be averaged on the whole batch, implementing it in the module needs the 'reduce' operation as in SyncBN. So I just write the line in the 'train.py', where the \mu from all GPUs are already together here.

**Comment by XiaLiPKU** [comment]
> And how to consider the gradient backpropgation in your implement?

For the second question:

I simple cut off the gradients for the A_E and A_M iterations as https://github.com/XiaLiPKU/EMANet/blob/9a492d8aaad297e15eac044b3bb9583e63ffa3a3/network.py#L227.
To be honest, there lacks deep exploration of what happens inside the EMA. So EMANet is just a naive exploration on the EM + Attention mechanism. So, I just look forward for more deep analysis by dear followers.
